### ISSUE TITLE
Script with prebid Server (S2S) and conditional ad size mapping unit doesn't works

### ISSUE THREAD

**Issue by salpure** [prebid/Prebid.js#2890]
## Type of issue
bug

## Description
Same issue as described here:  https://github.com/prebid/Prebid.js/issues/2222

## Steps to reproduce
https://github.com/prebid/Prebid.js/issues/2222

## Test page

```
<html>
  <head>
      <script>
        var PREBID_TIMEOUT = 2700;

        var pbjs = pbjs || {};
        pbjs.que = pbjs.que || [];

        var desktopSizes = [[970, 250], [728, 90], [160, 600], [300, 600], [300, 250], [1, 1]];
        var tabletSizes = [[728, 90], [160, 600], [300, 600], [300, 250], [1, 1]];
        var phoneSizes = [[320, 100], [320, 50], [300, 250], [1, 1]];

        var adUnits = [
            {
                code: 'Div1',
                mediaTypes: {
                  banner: {
                    sizes: [[300, 250], [970, 90], [728, 90], [970,250]]
                  }
                },
                bids: [
                    {
                        bidder: 'pubmatic',
                        labelAll: ['phone'],
                        params: {
                            publisherId: '123',
                            adSlot: '/dfp/slot1@300x250:0',
                            size: [300, 250],
              						  pmzoneid: 'zone1',
              							kadpageurl:'www.yahoo.com'
                        }
                    },
                    {
                        bidder: 'pubmatic',
                        labelAny: ['display', 'tablet'],
                        params: {
                            publisherId: '123',
                            adSlot: '/dfp/slot2@728x90:0',
              						  pmzoneid: '123123',
                            size: [728, 90],
              							kadpageurl:'www.yahoo.com'
                        }
                    },
                    {
                       bidder: 'appnexus',
                       labelAll: ['phone'],
                       params: {
                         placementId: '123123', // banner
                       }
                     },
                     {
                        bidder: 'appnexus',
                        labelAny: ['display', 'tablet'],
                        params: {
                          placementId: '123123', // banner
                        }
                      }
			          ]
            }
        ];


      </script>

      <script type="text/javascript" src="prebid1.16.0.js" async></script>
      <script>
        var googletag = googletag || {};
        googletag.cmd = googletag.cmd || [];

        pbjs.que.push(function () {
          pbjs.setConfig({
  					userSync: {
  						iframeEnabled: true,
  						enabledBidders: ['pubmatic'],
  				    syncDelay: 6000 //todo: default is 3000 write image pixels 5 seconds after the auction
  				  },
				  
        s2sConfig: {
            accountId: '1',
            enabled: true,
            bidders: ['appnexus', 'pubmatic'],
            timeout: 1000,
            adapter: 'prebidServer',
            endpoint: 'https://prebid.adnxs.com/pbs/v1/openrtb2/auction',
            syncEndpoint: 'https://prebid.adnxs.com/pbs/v1/cookie_sync',
            cookieSet: true,
            cookiesetUrl: 'https://acdn.adnxs.com/cookieset/cs.js'
        },
                    sizeConfig: [{
                        'mediaQuery': '(min-width: 1025px)',
                        'sizesSupported': desktopSizes,
                        'labels': ['desktop']
                    }, {
                        'mediaQuery': '(min-width: 768px) and (max-width: 1024px)',
                        'sizesSupported': tabletSizes,
                        'labels': ['tablet']
                    }, {
                        'mediaQuery': '(min-width: 320px) and (max-width: 812px)',
                        'sizesSupported': phoneSizes,
                        'labels': ['phone']
                    }]
          })
        });

        pbjs.que.push(function() {
          pbjs.addAdUnits(adUnits);
          pbjs.logging = true;
          pbjs.requestBids({
            bidsBackHandler: sendAdserverRequest
          });
        });

        function sendAdserverRequest(bids) {
          if (pbjs.adserverRequestSent) return;
          pbjs.adserverRequestSent = true;
          console.log("BIds aa gaye:", bids);
          googletag.cmd.push(function() {
            pbjs.que.push(function() {
              pbjs.setTargetingForGPTAsync();
              googletag.pubads().refresh();
            });
          });
        }

        setTimeout(function() {
          sendAdserverRequest();
        }, PREBID_TIMEOUT);

        pbjs.que.push(function () {
          pbjs.onEvent('bidResponse', function (bids) {
            console.log("BIds aa gaye:", bids);
          })
        });

      </script>

      <script>
        (function () {
          var gads = document.createElement('script');
          gads.async = true;
          gads.type = 'text/javascript';
          var useSSL = 'https:' == document.location.protocol;
          gads.src = (useSSL ? 'https:' : 'http:') +
            '//www.googletagservices.com/tag/js/gpt.js';
          var node = document.getElementsByTagName('script')[0];
          node.parentNode.insertBefore(gads, node);
        })();
      </script>

      <script>
        googletag.cmd.push(function () {
         googletag.defineSlot('/dfp/slot1', [[300, 250],[728, 90],[160, 600]], 'Div1').addService(googletag.pubads());
          // googletag.pubads().enableSingleRequest();
          googletag.pubads().disableInitialLoad();
          googletag.enableServices();
        });
      </script>
  </head>

  <body>
    <h2>Prebid.js Test</h2>
    <h5>pubmatic_test2</h5>
    <div id='Div1'>
        <script type='text/javascript'>
          googletag.cmd.push(function() {
      			googletag.display('Div1');
          });
        </script>
    </div>
  </body>
  </html>
```

### Expected results
Based on viewport/size the right size mapping should be applied.

### Actual results
Desktop size is applied even for mobile size viewport.  However, anything smaller than mobile size then warning is seen.  "WARNING: callBids executed with no bidRequests.  Were they filtered by labels or sizing? "

## Platform details
prebid1.16.0.js 
Prebid Server

## Other information
https://github.com/prebid/Prebid.js/issues/2222

**Comment by jsnellbaker** [comment]
Hello @salpure Thanks for submitting this issue.  I'll take a look into it.

**Comment by jsnellbaker** [comment]
Hi @salpure 

There's an existing PR that will address the filtering that should take place properly for s2s requests.  This PR is https://github.com/prebid/Prebid.js/pull/2795

In addition though, there is a setup issue in your labels.  The `labelAny` values in the `adUnits.bids` are using `display` whereas the corresponding label in the `sizeConfig` is named `desktop`.  You'll want to rectify one of these values so they're matching up correctly.

**Comment by jsnellbaker** [comment]
Hi @salpure,

The above mentioned PR was merged into master and included as part of the 1.18.0 release.  If you continue to see any additional trouble here, please let us know.
